plan/logical_plan_test.go

```diff
@@ -601,6 +601,11 @@ func (s *testPlanSuite) TestSubquery(c *C) {
 			sql:  "select * from t where exists (select s.a from t s having sum(s.a) = t.a )",
 			best: "Join{DataScan(t)->DataScan(s)->Aggr(sum(s.a))->Projection}->Projection",
 		},
+		{
+			// Test MaxOneRow for limit.
+			sql:  "select (select * from (select b from t limit 1) x where x.b = t1.b) from t t1",
+			best: "Join{DataScan(t1)->DataScan(t)->Projection->Limit}(t1.b,x.b)->Projection->Projection",
```

Comment:
How can we tell there is a MaxOneRow attribute in the plan?

Reply:
Before this test, the plan contains an Apply op.